added: recentf-remove-sudo-tramp-prefix

ncaq opened the pull request:
### Brief summary of what the package does

recentf save duplicate history, for instance.
"/sudo:root@akaza:/usr/share/emacs/24.5/lisp/net/tramp.el"
"/usr/share/emacs/24.5/lisp/net/tramp.el"
This Package normalise to after.

### Direct link to the package repository

<https://github.com/ncaq/recentf-remove-sudo-tramp-prefix>

### Your association with the package

maintainer

### Relevant communications with the upstream package maintainer

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

purcell commented:
This doesn't actually do what the `Commentary` text says: it will remove items from `recentf-list` unconditionally, even if they are not duplicates in the list.

Apart from that, the `.elc` file(s) here also need to be removed from the repo.

You might consider using `cl-lib`'s `cl-remove-if` function to simplify this code.

tarsius commented:
Closing since "this doesn't actually do what the Commentary text says" and since op did not respond.

ncaq commented:
I'm sorry.
I was busy.
I will fix it and start over again.

tarsius commented:
Okay! No problem 😄

ncaq commented:
I fixed `Commentary`.
I remove `.elc` file of git.

`cl-remove-if` case, I use recentf standard remove duplicate function.
I did not rewrite it because I do not know how to easily rewrite to a format that uses cl-remove-if.

I changed use minor-mode.

Can you accept it?

tarsius commented:
Lets wait for @purcell to do another review.

purcell commented:
Looks good, just pending https://github.com/ncaq/recentf-remove-sudo-tramp-prefix/pull/2.

ncaq commented:
I merged.

purcell commented:
> I merged.

Great, me too!

ncaq commented:
thanks!
I will try #5170 next!
